In [2]:
from youtube_transcript_api import YouTubeTranscriptApi

def get_subtitles(video_id):
    """Fetches the subtitles for a given YouTube video id."""
    try:
        transcript = YouTubeTranscriptApi.get_transcript(video_id='kzXJTb7MeRs')
        return transcript
    except Exception as e:
        print(f"An error occurred: {e}")
        return None

video_id = "KYxLEDF6kjs"
subtitles = get_subtitles(video_id)
print(subtitles)
if subtitles:
    for subtitle in subtitles:
        print(subtitle)
        break
    
        print(f"{subtitle['start']} - {subtitle['start'] + subtitle['duration']}: {subtitle['text']}")


[{'text': 'hello and welcome to this video in this', 'start': 0.68, 'duration': 3.52}, {'text': "video we'll explore the best visual", 'start': 2.8, 'duration': 3.36}, {'text': 'studio code extension which will help', 'start': 4.2, 'duration': 4.28}, {'text': 'help you work better with your', 'start': 6.16, 'duration': 5.599}, {'text': 'programming uh and productivity tool and', 'start': 8.48, 'duration': 6.159}, {'text': 'it will also help you with uh make it', 'start': 11.759, 'duration': 5.961}, {'text': 'make your vs code experience very much', 'start': 14.639, 'duration': 4.761}, {'text': 'more productive okay there are so many', 'start': 17.72, 'duration': 3.04}, {'text': "helpful extensions we're going to look", 'start': 19.4, 'duration': 3.92}, {'text': 'at this amazing extensions of 2024 so', 'start': 20.76, 'duration': 4.12}, {'text': 'these are the ones we are going to run', 'start': 23.32, 'duration': 4.32}, {'text': 'through uh the first one is tab n code', 'start': 24.88,

[
   {
      "text":"hello and welcome to this video in this",
      "start":0.68,
      "duration":3.52
   },
   {
      "text":"video we'll explore the best visual",
      "start":2.8,
      "duration":3.36
   },
   {
      "text":"studio code extension which will help",
      "start":4.2,
      "duration":4.28
   },
   {
      "text":"help you work better with your",
      "start":6.16,
      "duration":5.599
   },
   {
      "text":"programming uh and productivity tool and",
      "start":8.48,
      "duration":6.159
   },
   {
      "text":"it will also help you with uh make it",
      "start":11.759,
      "duration":5.961
   },
   {
      "text":"make your vs code experience very much",
      "start":14.639,
      "duration":4.761
   },
   {
      "text":"more productive okay there are so many",
      "start":17.72,
      "duration":3.04
   },
   {
      "text":"helpful extensions we're going to look",
      "start":19.4,
      "duration":3.92
   },
   {
      "text":"at this amazing extensions of 2024 so",
      "start":20.76,
      "duration":4.12
   },
   {
      "text":"these are the ones we are going to run",
      "start":23.32,
      "duration":4.32
   },
   {
      "text":"through uh the first one is tab n code",
      "start":24.88,
      "duration":5.799
   },
   {
      "text":"Runner live server draw.io preer",
      "start":27.64,
      "duration":5.439
   },
   {
      "text":"code spell checker Excel viewer to-do",
      "start":30.679,
      "duration":4.641
   },
   {
      "text":"and",
      "start":33.079,
      "duration":2.241
   },
   {
      "text":"material so let's head over to our uh",
      "start":35.96,
      "duration":6.04
   },
   {
      "text":"Visual Studio code we I have opened the",
      "start":39.68,
      "duration":4.48
   },
   {
      "text":"simple JavaScript folder okay we got",
      "start":42.0,
      "duration":4.719
   },
   {
      "text":"some code over here so let's look at the",
      "start":44.16,
      "duration":4.919
   },
   {
      "text":"first extension which is tab n so go to",
      "start":46.719,
      "duration":3.68
   },
   {
      "text":"the extensions",
      "start":49.079,
      "duration":5.521
   },
   {
      "text":"tab uh and search for tab",
      "start":50.399,
      "duration":4.201
   },
   {
      "text":"n okay at the top you see this uh dab 9",
      "start":55.239,
      "duration":5.681
   },
   {
      "text":"a auto complete and chat for JavaScript",
      "start":58.64,
      "duration":3.96
   },
   {
      "text":"python typescript go Java and more so",
      "start":60.92,
      "duration":3.919
   },
   {
      "text":"this are like what this basically does",
      "start":62.6,
      "duration":4.04
   },
   {
      "text":"it's a brilliant AI coding assistant",
      "start":64.839,
      "duration":3.441
   },
   {
      "text":"with AI code completion so this helps",
      "start":66.64,
      "duration":3.2
   },
   {
      "text":"you like you know if you're a beginner",
      "start":68.28,
      "duration":2.839
   },
   {
      "text":"or even an experience like if you want",
      "start":69.84,
      "duration":2.56
   },
   {
      "text":"to write faster code you know your",
      "start":71.119,
      "duration":3.64
   },
   {
      "text":"syntax to avoid syntax errors it will do",
      "start":72.4,
      "duration":3.96
   },
   {
      "text":"some auto complete for you which is",
      "start":74.759,
      "duration":5.201
   },
   {
      "text":"absolutely great so let's take a look",
      "start":76.36,
      "duration":5.719
   },
   {
      "text":"let's first install",
      "start":79.96,
      "duration":5.76
   },
   {
      "text":"it so it's going to take you to this",
      "start":82.079,
      "duration":5.921
   },
   {
      "text":"uh page which will say like you know",
      "start":85.72,
      "duration":6.12
   },
   {
      "text":"describes what it is about",
      "start":88.0,
      "duration":3.84
   },
   {
      "text":"so it's going to take some time so I",
      "start":92.759,
      "duration":4.64
   },
   {
      "text":"will just move ahead and in this video",
      "start":94.439,
      "duration":5.601
   },
   {
      "text":"to where the tab 9 is already",
      "start":97.399,
      "duration":5.04
   },
   {
      "text":"installed okay so it's getting installed",
      "start":100.04,
      "duration":4.64
   },
   {
      "text":"over",
      "start":102.439,
      "duration":2.241
   },
   {
      "text":"here so you got to accept some uh",
      "start":107.079,
      "duration":4.68
   },
   {
      "text":"settings you can also change it I'm",
      "start":109.799,
      "duration":5.241
   },
   {
      "text":"going to accept all settings for",
      "start":111.759,
      "duration":3.281
   },
   {
      "text":"now and I think it's pretty much it's",
      "start":117.68,
      "duration":5.24
   },
   {
      "text":"still initializing",
      "start":120.24,
      "duration":5.28
   },
   {
      "text":"it all right so now let's take a look at",
      "start":122.92,
      "duration":5.959
   },
   {
      "text":"this uh extension all right you will see",
      "start":125.52,
      "duration":5.719
   },
   {
      "text":"the tab n it says Get U code to get the",
      "start":128.879,
      "duration":3.641
   },
   {
      "text":"code suggestion as you code you will see",
      "start":131.239,
      "duration":3.0
   },
   {
      "text":"the tab n code suggestions in line in",
      "start":132.52,
      "duration":5.84
   },
   {
      "text":"Gray in addition to the IDS popup",
      "start":134.239,
      "duration":6.08
   },
   {
      "text":"suggestions all right so let's take a",
      "start":138.36,
      "duration":3.12
   },
   {
      "text":"look",
      "start":140.319,
      "duration":3.601
   },
   {
      "text":"here so we got to sign in okay we also",
      "start":141.48,
      "duration":6.32
   },
   {
      "text":"have to do a sign so let's do",
      "start":143.92,
      "duration":3.88
   },
   {
      "text":"that",
      "start":148.76,
      "duration":4.64
   },
   {
      "text":"so you got to go to this",
      "start":151.08,
      "duration":4.519
   },
   {
      "text":"website so what I'm going to do here is",
      "start":153.4,
      "duration":4.68
   },
   {
      "text":"I'm going to log in with my Google",
      "start":155.599,
      "duration":4.64
   },
   {
      "text":"account which is first easier you can",
      "start":158.08,
      "duration":4.64
   },
   {
      "text":"also give a",
      "start":160.239,
      "duration":2.481
   },
   {
      "text":"email so we're going to select my login",
      "start":162.879,
      "duration":3.801
   },
   {
      "text":"over here and we're going to proceed",
      "start":165.4,
      "duration":4.919
   },
   {
      "text":"into tab 9 it's going to ask me for some",
      "start":166.68,
      "duration":7.04
   },
   {
      "text":"password please go",
      "start":170.319,
      "duration":3.401
   },
   {
      "text":"ahead so now it tells me that you can",
      "start":178.76,
      "duration":3.92
   },
   {
      "text":"close this tab and start working with",
      "start":181.2,
      "duration":3.16
   },
   {
      "text":"this ID all right so awesome okay so",
      "start":182.68,
      "duration":4.6
   },
   {
      "text":"let's close this tab let's go back to",
      "start":184.36,
      "duration":6.519
   },
   {
      "text":"our this thing and let's open up the",
      "start":187.28,
      "duration":3.599
   },
   {
      "text":"code all right so here now let me try to",
      "start":205.12,
      "duration":6.56
   },
   {
      "text":"write one script",
      "start":208.12,
      "duration":3.56
   },
   {
      "text":"so here if you",
      "start":211.72,
      "duration":2.64
   },
   {
      "text":"see this automatically gives me the next",
      "start":216.84,
      "duration":5.399
   },
   {
      "text":"suggestion okay you can either choose to",
      "start":220.2,
      "duration":3.08
   },
   {
      "text":"accept",
      "start":222.239,
      "duration":5.401
   },
   {
      "text":"it so now if I want to write like some",
      "start":223.28,
      "duration":7.239
   },
   {
      "text":"logic other logic over",
      "start":227.64,
      "duration":6.84
   },
   {
      "text":"here like if I go out of",
      "start":230.519,
      "duration":3.961
   },
   {
      "text":"this",
      "start":238.68,
      "duration":3.0
   },
   {
      "text":"now I can also use it to explain a",
      "start":243.4,
      "duration":3.88
   },
   {
      "text":"section of my code suppose I go here and",
      "start":245.2,
      "duration":5.0
   },
   {
      "text":"select I I select say explain code so",
      "start":247.28,
      "duration":4.84
   },
   {
      "text":"like if I go here it gives me the",
      "start":250.2,
      "duration":4.2
   },
   {
      "text":"options as to what all I can do I",
      "start":252.12,
      "duration":3.72
   },
   {
      "text":"selected the explain code option I",
      "start":254.4,
      "duration":4.44
   },
   {
      "text":"selected this block so let me see I'm",
      "start":255.84,
      "duration":5.44
   },
   {
      "text":"going to click",
      "start":258.84,
      "duration":2.44
   },
   {
      "text":"continue",
      "start":268.6,
      "duration":5.2
   },
   {
      "text":"initially it takes some time to",
      "start":271.6,
      "duration":4.76
   },
   {
      "text":"work so here it goes ahead and explains",
      "start":273.8,
      "duration":3.839
   },
   {
      "text":"to me like you know what that code is",
      "start":276.36,
      "duration":3.04
   },
   {
      "text":"doing which is really awesome okay and",
      "start":277.639,
      "duration":3.201
   },
   {
      "text":"it also gives",
      "start":279.4,
      "duration":4.32
   },
   {
      "text":"you the full details as to what",
      "start":280.84,
      "duration":4.28
   },
   {
      "text":"functions it is doing and what else it",
      "start":283.72,
      "duration":3.68
   },
   {
      "text":"is doing okay so we have this thing over",
      "start":285.12,
      "duration":4.68
   },
   {
      "text":"here which is I feel really helpful for",
      "start":287.4,
      "duration":6.6
   },
   {
      "text":"us so it will also explain your",
      "start":289.8,
      "duration":4.2
   },
   {
      "text":"code and now we can also say generate",
      "start":296.28,
      "duration":4.0
   },
   {
      "text":"test or we can write you know doc",
      "start":298.72,
      "duration":3.039
   },
   {
      "text":"strings for code or you can also fix",
      "start":300.28,
      "duration":4.12
   },
   {
      "text":"code now let's say suppose we have a",
      "start":301.759,
      "duration":5.321
   },
   {
      "text":"error over here okay we not we have a",
      "start":304.4,
      "duration":5.56
   },
   {
      "text":"syntax error over here now when we try",
      "start":307.08,
      "duration":5.72
   },
   {
      "text":"to save this it's going to give an error",
      "start":309.96,
      "duration":4.88
   },
   {
      "text":"so if I cover this part of the code and",
      "start":312.8,
      "duration":3.88
   },
   {
      "text":"say fix",
      "start":314.84,
      "duration":5.24
   },
   {
      "text":"code it takes some",
      "start":316.68,
      "duration":5.56
   },
   {
      "text":"time so it's telling me a couple of",
      "start":320.08,
      "duration":7.04
   },
   {
      "text":"Errors over here so it gives me the",
      "start":322.24,
      "duration":4.88
   },
   {
      "text":"example of how to fix this",
      "start":327.4,
      "duration":4.079
   },
   {
      "text":"so which is already great I can just fix",
      "start":332.56,
      "duration":4.44
   },
   {
      "text":"it from here or I can just say insert",
      "start":335.24,
      "duration":4.04
   },
   {
      "text":"and it replaces the whole thing all",
      "start":337.0,
      "duration":4.84
   },
   {
      "text":"right so you can make a judgment call",
      "start":339.28,
      "duration":4.32
   },
   {
      "text":"over here but what I'm saying over here",
      "start":341.84,
      "duration":4.04
   },
   {
      "text":"is this is applicable for Java Python",
      "start":343.6,
      "duration":4.56
   },
   {
      "text":"and it's extremely helpful so this is",
      "start":345.88,
      "duration":5.4
   },
   {
      "text":"one of my top tools uh which I suggest",
      "start":348.16,
      "duration":5.64
   },
   {
      "text":"especially in the world of AI now and",
      "start":351.28,
      "duration":4.44
   },
   {
      "text":"this is one of the best extensions that",
      "start":353.8,
      "duration":4.16
   },
   {
      "text":"is tab 9 which gives you AI suggestions",
      "start":355.72,
      "duration":4.84
   },
   {
      "text":"for your code",
      "start":357.96,
      "duration":4.4
   },
   {
      "text":"all",
      "start":360.56,
      "duration":4.639
   },
   {
      "text":"right next one the next extension we'll",
      "start":362.36,
      "duration":5.399
   },
   {
      "text":"look at code",
      "start":365.199,
      "duration":5.56
   },
   {
      "text":"Runner",
      "start":367.759,
      "duration":3.0
   }
]
